modules/afmongodb/CMakeLists.txt

```diff
@@ -0,0 +1,37 @@
+find_package(libmongoc-1.0)
+
+module_switch(ENABLE_MONGODB "Enable mongodb destination driver" libmongoc-1.0_FOUND)
+
+if (libmongoc-1.0_FOUND)
+
+  set(AFMONGODB_HEADERS
+    "afmongodb.h"
+    "afmongodb-parser.h"
+    "afmongodb-private.h"
+    "${CMAKE_CURRENT_BINARY_DIR}/afmongodb-grammar.h"
+  )
+
+  set(AFMONGODB_SOURCES
+    "afmongodb.c"
+    "afmongodb-parser.c"
+    "${CMAKE_CURRENT_BINARY_DIR}/afmongodb-grammar.c"
+  )
+
+set(AFMONGODB_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
+
+generate_y_from_ym(modules/afmongodb/afmongodb-grammar)
+bison_target(AFMongoDBGrammar
+  ${CMAKE_CURRENT_BINARY_DIR}/afmongodb-grammar.y
+  ${CMAKE_CURRENT_BINARY_DIR}/afmongodb-grammar.c
+  COMPILE_FLAGS ${BISON_FLAGS})
+
+include_directories (afmongodb SYSTEM PRIVATE ${MONGOC_INCLUDE_DIRS})
```

Comment:
So you are saying that I should replace `include_directories()` instead of fixing it?
(Or which `target_include` is _before_ `add_library`)?

Could you tell me what is the syslog-ng way of using cmake?  There are differencies between modules...

Reply:
> So you are saying that I should replace include_directories() instead of fixing it?

`target_include_directories()` is already in this file 2 lines above, that's why I asked to use it. I also think `target_*` is cleaner, because it is bound to a target, so it's not global.

> Could you tell me what is the syslog-ng way of using cmake? There are differencies between modules...

I don't know if there is a syslog-ng way. My review notes are based on the concept I described in #2007. This "modern CMake" thing helped to detect a few build issues and made build files a bit shorter and cleaner; but I don't want to force it to anyone.